portmgr/dmg/postflight.in

```diff
@@ -280,7 +295,17 @@ esac
 if /usr/bin/su "${USER}" -l -c "/usr/bin/printenv PATH" | tail -n 1 | tr ":" "\n" | grep "^${BINPATH}$" > /dev/null; then
     echo "Your shell already has the right PATH environment variable for use with MacPorts!"
 else
-    write_setting PATH "\"${BINPATH}:${SBINPATH}:\$PATH\""
+    if $IS_XONSH; then
+        # ↓"${ENV_COMMAND}${ENV_SEP}${1}${ASSIGN}${2}"
+        #  "$PATH.add('${PATH_TO_ADD}' ,front=True)"
+        backupASSIGN=${ASSIGN}
+        export ASSIGN=".add('"
+        write_setting PATH "${SBINPATH}',front=True)"
+        write_setting PATH "${BINPATH}' ,front=True)"
+        export ASSIGN=${backupASSIGN}
+    else
+        write_setting PATH "\"${BINPATH}:${SBINPATH}:\$PATH\""
+    fi
```

Comment:
I think it would make more sense to add the special handling for xonsh to the write_setting function and pass the shell's name as an additional parameter. This would avoid the duplication of this conditional block for each variable.

Reply:
that would replace some of the duplication with a new variable condition in the write_setting function, and you'd have to call this function 3 times instead of 2, so will add more lines with comments to user configs, so it seems even worse